content/en/blog/2021/migrate-alpha-security-policy/index.md

```diff
@@ -0,0 +1,273 @@
+    {{< /text >}}
+
+1. Apply the `PeerAuthentication`, `RequestAuthentication` and `AuthorizationPolicy` and monitor the traffic to make
+   sure it works as expected.
+
+## the `v1alpha1` RBAC policy migration
+
+The typical flow of migrating to `v1beta1` policy is to start by checking the `ClusterRbacConfig` to decide which
+namespace or service is enabled with RBAC.
+
+For each service enabled with RBAC:
+
+1. Get the workload selector from the service definition.
+
+1. Create a `v1beta1` policy with the workload selector.
+
+1. Update the `v1beta1` policy for each `ServiceRole` and `ServiceRoleBinding` applied to the service.
+
+1. Apply the `v1beta1` policy and monitor the traffic to make sure the policy is working as expected.
+
+1. Repeat the process for the next service enabled with RBAC.
+
+For each namespace enabled with RBAC:
+
+1. Apply a `v1beta1` policy that denies all traffic to the given namespace.
+
+## `v1alpha1` RBAC policy example
+
+Assume you have the following `v1alpha1` RBAC policies for the `httpbin` service in the `foo` namespace:
+
+{{< text yaml >}}
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ClusterRbacConfig
+metadata:
+  name: default
+spec:
+  mode: 'ON_WITH_INCLUSION'
+  inclusion:
+    namespaces: ["foo"]
+---
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRole
+metadata:
+  name: httpbin
+  namespace: foo
+spec:
+  rules:
+  - services: ["httpbin.foo.svc.cluster.local"]
+    methods: ["GET"]
+---
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: httpbin
+  namespace: foo
+spec:
+  subjects:
+  - user: "cluster.local/ns/default/sa/sleep"
+    roleRef:
+      kind: ServiceRole
+      name: "httpbin"
+{{< /text >}}
+
+Migrate the above RBAC policies to `v1beta1` in the following ways:
+
+1. Assume the `httpbin` service in the `foo` namespace has the following workload selector:
+
+    {{< text yaml >}}
+    selector:
+      app: httpbin
+      version: v1
+    {{< /text >}}
+
+1. Create the `AuthorizationPolicy` with the following content:
+
+    {{< text yaml >}}
+    apiVersion: security.istio.io/v1beta1
+    kind: AuthorizationPolicy
+    metadata:
+      name: httpbin
+      namespace: foo
+    spec:
+      selector:
+        matchLabels:
+          app: httpbin
+          version: v1
+      action: ALLOW
+      rules:
+      - from:
+        - source:
+            principals: ["cluster.local/ns/default/sa/sleep"]
+        to:
+        - operation:
+            methods: ["GET"]
+    {{< /text >}}
+
+1. Create the following `AuthorizationPolicy` that denies all traffic to the `foo` namespace because the `foo` namespace
+   was enabled with RBAC:
+
+    {{< text yaml >}}
+    apiVersion: security.istio.io/v1beta1
+    kind: AuthorizationPolicy
+    metadata:
+      name: allow-nothing
+      namespace: foo
+    spec:
+      {}
+    {{< /text >}}
+
+1. Apply the `AuthorizationPolicy` and monitor the traffic to make sure it works as expected.
```

Comment:
Do you plan to mention the conversion tool?

Reply:
Yes but I plan to add it in a follow up PR after I moved the tool under istio org.